scripts/download/tools.ts

```diff
@@ -0,0 +1,404 @@
+    linux:  ['docker-credential-secretservice', 'docker-credential-pass'],
+    darwin: ['docker-credential-osxkeychain'],
+    win32:  ['docker-credential-wincred'],
+  }[context.platform];
+  const promises = [];
+  const baseUrl = 'https://github.com/docker/docker-credential-helpers/releases/download';
+
+  for (const baseName of credHelperNames) {
+    const sourceUrl = `${ baseUrl }/v${ version }/${ baseName }-v${ version }-${ arch }.${ extension }`;
+    const binName = context.platform.startsWith('win') ? `${ baseName }.exe` : baseName;
+    const destPath = path.join(context.binDir, binName);
+
+    promises.push(downloadFunc(sourceUrl, destPath));
+  }
+
+  return Promise.all(promises);
+}
+
+/**
+ * Download the version of docker-credential-ecr-login for a specific platform.
+ * @param platform The platform we're downloading for.
+ * @param destDir The directory to place downloaded cred helper in.
+ */
+function downloadECRCredHelper(context: DownloadContext): Promise<void> {
+  const arch = context.isM1 ? 'arm64' : 'amd64';
+  const ecrLoginPlatform = context.platform.startsWith('win') ? 'windows' : context.platform;
+  const baseName = 'docker-credential-ecr-login';
+  const baseUrl = 'https://amazon-ecr-credential-helper-releases.s3.us-east-2.amazonaws.com';
+  const binName = context.platform.startsWith('win') ? `${ baseName }.exe` : baseName;
+  const sourceUrl = `${ baseUrl }/${ context.versions.ECRCredenialHelper }/${ ecrLoginPlatform }-${ arch }/${ binName }`;
+  const destPath = path.join(context.binDir, binName);
+
+  return download(sourceUrl, destPath);
+}
+
+export default async function downloadDependencies(rawPlatform: DependencyPlatform, depVersions: DependencyVersions): Promise<void> {
+  const platform = rawPlatform === 'wsl' ? 'linux' : rawPlatform;
+  const resourcesDir = path.join(process.cwd(), 'resources');
+  const downloadContext: DownloadContext = {
+    versions:          depVersions,
+    dependencyPlaform: rawPlatform,
+    platform,
+    kubePlatform:      platform === 'win32' ? 'windows' : platform,
+    isM1:              !!process.env.M1,
+    resourcesDir,
+    binDir:            path.join(resourcesDir, platform, 'bin'),
+    internalDir:       path.join(resourcesDir, platform, 'internal'),
+  };
+
+  fs.mkdirSync(downloadContext.binDir, { recursive: true });
+  fs.mkdirSync(downloadContext.internalDir, { recursive: true });
+  fs.mkdirSync(path.join(downloadContext.resourcesDir, 'linux', 'internal'), { recursive: true });
+
+  await Promise.all([
+    downloadKuberlrAndKubectl(downloadContext),
+    downloadHelm(downloadContext),
+    downloadDockerCLI(downloadContext),
+    downloadDockerBuildx(downloadContext),
+    downloadDockerCompose(downloadContext),
+    downloadTrivy(downloadContext),
+    downloadSteve(downloadContext),
+    downloadGuestAgent(downloadContext),
+    downloadRancherDashboard(downloadContext),
+    downloadDockerProvidedCredHelpers(downloadContext),
```

Comment:
This downloads `docker-credential-pass` / `docker-credential-secretservice` on WSL; is that intended?

Reply:
Same comment as https://github.com/rancher-sandbox/rancher-desktop/pull/2725#discussion_r947088944.